packages/components/pie-modal/test/visual/pie-modal.spec.ts

```diff
@@ -1,133 +1,128 @@
-import { test, expect } from '@sand4rt/experimental-ct-web';
+import { test, expect } from '@playwright/test';
 import { percyWidths } from '@justeattakeaway/pie-webc-testing/src/percy/breakpoints.ts';
 import percySnapshot from '@percy/playwright';
-import { PieModal } from '../../src/index.ts';
 import { type ModalProps, sizes, positions } from '../../src/defs.ts';
-
-const componentSelector = '[data-test-id="pie-modal"]';
-const componentFooterSelector = '[data-test-id="pie-modal-footer"]';
+import { ModalDefaultPage } from '../helpers/page-object/pie-modal-default.page.ts';
+import { ModalLargeTextContentPage } from '../helpers/page-object/pie-modal-large-text-content.page.ts';
 
 sizes.forEach((size) => {
-    test(`should render correctly with size = ${size}`, async ({ page, mount }) => {
-        await mount(PieModal, {
-            props: {
-                heading: 'This is a modal heading',
-                isOpen: true,
-                size,
-                leadingAction: {
-                    text: 'Confirm',
-                },
-            } as ModalProps,
-        });
-
-        const modal = page.locator(componentSelector);
-        await expect.soft(modal).toBeVisible();
+    test(`should render correctly with size = ${size}`, async ({ page }) => {
+        const modalDefaultPage = new ModalDefaultPage(page);
+        const props: Partial<ModalProps> = {
```

Comment:
Do we need `Partial`? Most properties are already optional, and if there is anything that is required we should be setting it in the tests anyway

Reply:
Are you happy for this to be covered as part of the post-refactor work?

Reply:
`heading` is a required prop with a default value set in Storybook that for most tests, we don't modify. So it's either continue using `Partial` or set `heading` to the same value that's set by the Storybook default to prevent further visual changes.

Reply:
Setting the `heading` sounds good to me. It looks like they might all share the same one so you could always pull it out into a  variable